src/main/scala/inox/solvers/z3/Z3Native.scala

```diff
@@ -501,50 +514,28 @@ trait Z3Native extends ADTManagers with Interruptible { self: AbstractSolver =>
       val kind = z3.getASTKind(t)
       kind match {
         case Z3NumeralIntAST(Some(v)) =>
-          val leading = t.toString.substring(0, 2 min t.toString.length)
```

Comment:
After toying a bit with this to fix some bugs, it can be noticeably simplified to this patch.